packages/match-media/src/useBreakpointIndex.js

```diff
@@ -0,0 +1,35 @@
+import { useState, useEffect, useCallback } from 'react'
+import { useThemeUI } from 'theme-ui'
+
+const useBreakpointIndex = () => {
+  const {
+    theme: { breakpoints },
+  } = useThemeUI()
+
+  const getIndex = useCallback(() => {
+    let index = breakpoints.findIndex(
+      breakpoint => window.matchMedia(`(max-width: ${breakpoint})`).matches
```

Comment:
These should be `min-width` right? The `0` in responsive array values corresponds to no media query, so maybe it'd make sense to pad the `theme.breakpoints` array with an item for that one if this uses `findIndex` – I suspect mixing `min-width` and `max-width` would yield different results when viewport is at exactly the width of the breakpoint (e.g. at `640px` viewport when `breakpoints = [ '40em', '52em', '64em' ]`)

Reply:
Yes you're right, there is a px difference. I've made an update to address this, and it simplifies the logic too.

Reply:
Here's a demo https://codesandbox.io/s/theme-uimatch-media-tzv1s